### PR TITLE
Bulk resolve

### PR DESCRIPTION
**Pull request by mchrapek**

### CODE REVIEW

File: src/main/scala/org/constellation/p2p/DataResolver.scala [hunk 1]
```diff
@@ -257,6 +257,39 @@ class DataResolver extends StrictLogging {
...
+      val requestHashes = allHashes.filterNot(innerHashes.toSet)
+
+      getBatchData[T](requestHashes, endpoint, peer)(contextToReturn).flatMap { data =>
+        makeAttempt(filteredPeers, allHashes, innerHashes ++ data.map(_._1), response ++ data.map(_._2))
+      }
+    }
+
+    for {
+      t <- makeAttempt(sortedPeers, hashes)
+    } yield t
```
Comment (marcinwadon) [inline src/main/scala/org/constellation/p2p/DataResolver.scala]: just `makeAttempt(sortedPeers, hashes)`?

**Comment by mchrapek** [comment]
> There is no usage and handling partial responses = filtering resolved transactions and continuing queries to other peers when not all data has been received.

There is handling partial responses and continuing queries to other peers

File: src/main/scala/org/constellation/util/CommonEndpoints.scala [hunk 2]
```diff
@@ -125,6 +126,19 @@ trait CommonEndpoints extends Json4sSupport {
...
         APIDirective.handle(dao.transactionService.lookup(h))(complete(_))
       } ~
+      path("batch" / "transactions") {
+        parameter("ids") { ids =>
+          APIDirective.handle(
+            ids
+              .split(",")
+              .toList
+              .traverse(id => dao.transactionService.lookup(id).map((id, _)))
+              .flatMap(_.filter(_._2.isDefined).pure[IO])
```
Comment (marcinwadon) [inline src/main/scala/org/constellation/util/CommonEndpoints.scala]: why not `.map(_.filter(_._2.isDefined))`?

File: src/test/scala/org/constellation/p2p/PeerAPITest.scala [hunk 3]
```diff
@@ -242,6 +242,39 @@ class PeerAPITest
           }
         }
       }
+
+      "GET transactions" - {
+        "one of two transaction exist should return list of transactions" in {
```
Comment (marcinwadon) [inline src/test/scala/org/constellation/p2p/PeerAPITest.scala]: should return list of transactions if one of two transaction exists

File: src/test/scala/org/constellation/p2p/PeerAPITest.scala [hunk 4]
```diff
@@ -242,6 +242,39 @@ class PeerAPITest
...
+
+          val hashes = List("hash1", "none").mkString(",")
+
+          Get(s"/batch/transactions?ids=$hashes") ~> peerAPI.commonEndpoints ~> check {
+            status shouldEqual StatusCodes.OK
+            responseAs[List[(String, Option[TransactionCacheData])]].size shouldEqual 1
+          }
+        }
+
+        "transaction doesn't exist should return empty list" in {
```
Comment (marcinwadon) [inline src/test/scala/org/constellation/p2p/PeerAPITest.scala]: should return empty list if transactions are not exist

**Pull request merged**